stake/src/identity_key.rs

```diff
@@ -0,0 +1,55 @@
+use penumbra_crypto::rdsa::{SpendAuth, VerificationKey};
+use penumbra_proto::{serializers::bech32str, stake as pb, Protobuf};
+use serde::{Deserialize, Serialize};
+
+/// A [`SpendAuth`] [`VerificationKey`] used as a validator's identity key.
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(try_from = "pb::IdentityKey", into = "pb::IdentityKey")]
+pub struct IdentityKey(pub VerificationKey<SpendAuth>);
+
+impl std::str::FromStr for IdentityKey {
+    type Err = anyhow::Error;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        pb::IdentityKey {
+            ik: bech32str::decode(
+                s,
+                crate::VALIDATOR_IDENTITY_BECH32_PREFIX,
+                bech32str::Bech32m,
+            )?,
+        }
+        .try_into()
+    }
+}
+
+impl std::fmt::Display for IdentityKey {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&bech32str::encode(
+            &self.0.to_bytes(),
+            crate::VALIDATOR_IDENTITY_BECH32_PREFIX,
+            bech32str::Bech32m,
+        ))
+    }
+}
```

Comment:
not 100% happy with this, it feels somewhat cumbersome & duplicative -- it'd be neat to come up with a way to reuse the Serialize/Deserialize implementations directly...